### PR TITLE
cbor: fix off-by-one error

### LINKED ISSUE #5420: Off by one error in cbor_deserialize_float()
At line 524 in the current master of cbor.c at:

https://github.com/RIOT-OS/RIOT/blob/master/sys/cbor/cbor.c

The function returns 4, but as it needs to account for the preceding byte that flags a float value this should be 5.

You can see the same pattern at lines 549-550 for cbor_deserialize_double() where it returns 9 instead of 8.

[cbor.zip](https://github.com/RIOT-OS/RIOT/files/252810/cbor.zip)

```
size_t cbor_deserialize_float(const cbor_stream_t *stream, size_t offset, float *val)
{
    if (CBOR_TYPE(stream, offset) != CBOR_7 || !val) {
        return 0;
    }

    unsigned char *data = &stream->data[offset];

    if (*data == CBOR_FLOAT32) {
        *val = ntohf(*(uint32_t *)(data + 1));
        return 4;
    }

    return 0;
}
```

### PR DESCRIPTION
**Pull request by miri64**
Fixes #5420. I don't understand the unittests enough to see if they pick it up, and if yes, how to fix them. (or how to fix them that they *can* register such off by one errors, if they are not testing for it).

### CODE REVIEW

File: sys/cbor/cbor.c [hunk 1]
```diff
@@ -522,7 +522,7 @@ size_t cbor_deserialize_float(const cbor_stream_t *stream, size_t offset, float
 
     if (*data == CBOR_FLOAT32) {
         *val = ntohf(*(uint32_t *)(data + 1));
-        return 4;
+        return 5;
```
Comment (cgundogan) [inline sys/cbor/cbor.c]: the [doc](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/cbor.h#L335) clearly states that it should be the number of bytes written into `val`. I don't see how `5` can be correct here?

Comment (miri64) [inline sys/cbor/cbor.c]: It includes also the flags field (as #5420 points out). If the 4 would be correct the 9 below in ll. 549-550 would be incorrect.

Comment (miri64) [inline sys/cbor/cbor.c]: (as would be the implementation of `encode_int()` btw).

**Comment by miri64** [comment]
Fixed doc to better reflect return value.

**Comment by miri64** [comment]
Squashed.

**Comment by miri64** [comment]
Please re-ACK, I've adapted the unittests to detect this kind of error.

**Comment by cgundogan** [comment]
re-ACK

**Comment by x3ro** [comment]
Merging in accordance with @cgundogan

**Pull request merged**